compressed_image_transport/src/compressed_publisher.cpp

```diff
@@ -78,31 +78,22 @@ void CompressedPublisher::configCb(Config& config, uint32_t level)
 
 void CompressedPublisher::publish(const sensor_msgs::Image& message, const PublishFn& publish_fn) const
 {
-  // Compressed image message
-  sensor_msgs::CompressedImage compressed;
-  compressed.header = message.header;
-  compressed.format = message.encoding;
-
-  // Compression settings
-  std::vector<int> params;
-
   // Get codec configuration
   compressionFormat encodingFormat = UNDEFINED;
   if (config_.format == compressed_image_transport::CompressedPublisher_jpeg)
     encodingFormat = JPEG;
   if (config_.format == compressed_image_transport::CompressedPublisher_png)
     encodingFormat = PNG;
 
-  // Bit depth of image encoding
-  int bitDepth = enc::bitDepth(message.encoding);
-  int numChannels = enc::numChannels(message.encoding);
-
+  // Compression settings
+  std::vector<int> params;
+  sensor_msgs::CompressedImagePtr compressed;
   switch (encodingFormat)
   {
     // JPEG Compression
     case JPEG:
     {
-      params.resize(9, 0);
+      params.resize(8, 0);
```

Comment:
Also here, why did that number change?

Reply:
It is the size of the local vector. Which is eight and not nine if you look at the lines below where the vector is filled up; Furthermore it is just plain wrong to have an odd number as the size of this container since it needs to have pairs of OpenCV flags and values. As stated by the documentation the params vector contains _"Format-specific parameters encoded as pairs"_; (see docu [here](https://docs.opencv.org/3.4/d4/da8/group__imgcodecs.html#gabbc7ef1aa2edfaa87772f1202d67e0ce) )

Reply:
OK. So, also a bug-fix. Good.